### ISSUE TITLE
Validate dictionary members do not shadow members of any inherited dictionary

### ISSUE THREAD

**Issue by foolip** [w3c/webidl2.js#580]
https://heycam.github.io/webidl/#idl-dictionaries says:

> The identifier of a dictionary member must not be the same as that of another dictionary member defined on the dictionary or on that dictionary’s inherited dictionaries.

I tried if webidl2.js would catch such errors but it seems like not:
```js
const { parse, validate } = require("webidl2");
const ast = parse(`dictionary A { any m; }; dictionary B : A { object m; };`);
const validations = validate(ast); // returns []
```

**Comment by foolip** [comment]
It actually looks like duplicating the member name on a single dictionary is also not validated for.

**Comment by foolip** [comment]
I wrote up a quick check for this locally (in webref) but to my surprise there doesn't seem to be any duplicate member names currently.

**Comment by saschanaz** [comment]
Yeah, the current name check only applies for parent types, not members. Something to be done later.
